### DISCUSSION TITLE
Adding data to a trial retrospectively

### DISCUSSION THREAD

**Discussion by yuvalharr** [jspsych/jsPsych#1051] [Q&A] [answered]
Hey there,

My experiment goes like this: there is a task-trial (let's call it trial 1) and right after it two check-trials (2 and 3) to check if the subject answered correctly regarding trial 1. If both trials 2 and 3 are correct- I can conclude that trial 1 was successful.

My question is this:
Is there anyway I can add data to trial 1 only after trials 2 and 3 are finished? In this way I would be able to add an attribute to trial 1 - Success= true or Success=false - depending on trials 2 and 3 results.

The closest thing I found was this function '''jsPsych.data.get().addToLast({success: true});''' , but it doesn't seem to allow adding data to a trial earlier then the last one.

Any help would be great! 
Thanks,
Yuval

**Comment by becky-gilbert** [comment, marked as answer]
I think the `addToLast` function will add a set of properties to the last trial _in the data collection_, which means you could use the [`filter` function](https://www.jspsych.org/core_library/jspsych-data/#filter) first to select all of your task trials, then use the `addToLast` function to add data to the most recent trial of that type. You would need a key-value pair in your task trial data that you can filter on, e.g. `my_trial_type: "task-trial"`, then you could do this:
```javascript
jsPsych.data.get().filter({my_trial_type: "task-trial"}).addToLast({success: true});  
```
Or, instead of using `addToLast`, you could use the [`values` function](https://www.jspsych.org/core_library/jspsych-data/#values), because that gives you a modifiable array. 
```javascript
var trial_1_data = jsPsych.data.get().filter({my_trial_type: "task-trial"}).values();
// index the last element in the array, which is the most recent trial that meets the filter criteria
trial_1_data[trial_1_data.length-1].success = true;
```
Also, if use the `values` method, you could get all of the data (i.e. skip the filter step) and then index the array to get back to the last task trial. I think it would look like this:
```javascript
var all_data = jsPsych.data.get().values();
// go back 2 trials, which means subtracting 2 from the array length, and 1 more to account for the 0 indexing
all_data[all_data.length-3].success = true; 
```
I haven't tested this, so please let us know if any of these options work!

**Reply by yuvalharr** [reply]
Tried your third suggestion- works perfectly well!

Thank you so much for your time and effort!
